docs/src/cookbooks/verilog-vs-chisel.md

```diff
@@ -0,0 +1,920 @@
+  val io = IO(new Bundle {
+    val in0 = Input(UInt(in0Width.W))
+    val in1 = Input(UInt(in1Width.W))
+    val sum = Output(UInt(sumWidth.W))
+  })
+  // a +& b includes the carry, a + b does not
+  io.sum := io.in0 +& io.in1
+}
+```
+</td>
+</tr>
+<tr>
+<td>
+
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+# Wire Assignment and Literal Values
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+wire [31:0] 
+a = 32'd42; 
+wire [31:0] 
+b = 32'hbabecafe; 
+wire [15:0] c; 
+assign c = 16'b1;
+```
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule extends Module {
+ val a = WireDefault(42.U(32.W))
+ val b = WireDefault("hbabecafe".U(32.W))  
+ val c = Wire(UInt(16.W)) 
+  c := "b1".U;
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule)
+```
+</td>
+         </tr>
+<tr>
+<td>
+
+```verilog
+module MyWireAssignmentModule2 ();
+
+wire [31:0] aa = 'd42;
+ reg [31:0] a; // Logical register for use in always block, not real register
+ 
+ //
+always @(*) begin
+  a = aa;
+end 
+
+// Hex value initialization
+ wire [31:0] b = 32'hbabecafe;
+
+ // Declaration separate from Assignment
+ wire [15:0] c;
+ wire d;
+
+ assign c = 16'b1;
+ assign d = 1'b1;
+
+// Signed values
+wire signed [63:0] g;
+assign g = -’d5;
+
+wire signed [31:0] h = 'd5;
+
+reg signed[31:0] f;
+always@(*) begin
+  f = ‘d5;
+end
+endmodule
+```
+
+
+</td>
+<td>
+
+```scala mdoc:silent
+
+
+class MyWireAssignmentModule2 extends Module {
+
+val aa = 42.U(32.W)
+val a = Wire(UInt(32.W))
+a := aa
+val b = "hbabecafe".U(32.W)
+val c = Wire(UInt(16.W))
+val d = Wire(Bool())
+c := "b1".U(16.W)
+d := true.B
+val g = Wire(SInt(64.W))
+g := -5.S
+val h = 5.asSInt(32.W)
+val f = Wire(SInt(32.W))
+f := 5.S
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new MyWireAssignmentModule2)
+```
+
+</td>
+</tr>
+    </table>
+<html>
+<body>
+
+# Register Declaration and Assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+         </tr>
+         <tr>
+<td>
+
+```verilog
+module RegisterModule(
+  input         clock,
+  input         reset,
+  input  [11:0] io_in,
+  output [11:0] io_out
+);
+  assign io_out = io_in; // @[main.scala 18:10]
+endmodule
+```
+</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(12.W))
+    val out = Output(UInt(12.W))
+  })
+
+  val registerWithInit = RegInit(42.U(12.W))
+  registerWithInit := registerWithInit - 1.U
+  io.out := io.in
+}
+```
+```scala mdoc:invisible
+ChiselStage.emitVerilog(new RegisterModule)
+```
+</td>
+         </tr>
+ <tr>
+<td>
+
+```verilog
+module RegisterModule(
```

Comment:
For this one, again let's see if we can adjust the whitespace on either side so the related concepts show up more side-by-side